src/components/Header/Header.tsx

```diff
@@ -0,0 +1,67 @@
+import cn from 'classnames';
+import { useContext, useEffect, useRef } from 'react';
+import { ErrorType } from '../../types/ErrorType';
+import { AppContext } from '../../contexts/appContext';
+
+interface Props{
+  isEveryTodosCompleted: boolean,
+}
+
+export const Header: React.FC<Props> = ({ isEveryTodosCompleted }) => {
+  const {
+    todoTitle,
+    setTodoTitle,
+    createNewTodo,
+    setErrorMessage,
+    isLoading,
+    displayError,
+  } = useContext(AppContext);
+
+  const todoTitleRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (todoTitleRef.current && !todoTitleRef.current.disabled) {
+      todoTitleRef.current.focus();
+    }
+  });
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+    if (!todoTitle.trim()) {
+      displayError(ErrorType.TitleIsEmpty);
+
+      setTimeout(() => setErrorMessage(null), 2000);
```

Comment:
Why? If you handle removing the error message inside `ErrorInfo` component, and the delay is different...